src/wallet/rpcwallet.cpp

```diff
@@ -709,9 +709,18 @@ static UniValue getbalance(const JSONRPCRequest& request)
 
     LOCK2(cs_main, pwallet->cs_wallet);
 
-    const UniValue& dummy_value = request.params[0];
-    if (!dummy_value.isNull() && dummy_value.get_str() != "*") {
-        throw JSONRPCError(RPC_METHOD_DEPRECATED, "dummy first argument must be excluded or set to \"*\".");
+    const UniValue& trusted_only_uv = request.params[0];
+    bool trusted_only = true;
+    if (trusted_only_uv.isBool()) {
```

Comment:
I think we should try to avoid having a single argument accept two different types. It's messy and I don't think it's supported by the `CRPConvertTable` logic.

Reply:
It's only a boolean. The string stuff is strictly for backward compatibility.

Reply:
`bitcoin-cli` is just a testing tool. As such, I don't think we need to worry about backward compatibility there.

This is fully backward compatible with any normal JSON-RPC client.